v3/src/components/calculator/calculator-title-bar.tsx

```diff
@@ -6,11 +6,12 @@ import MinimizeIcon from "../../assets/icons/icon-minimize.svg"
 import { ITileTitleBarProps } from "../tiles/tile-base-props"
 
 export const CalculatorTitleBar = ({tile, onCloseTile}: ITileTitleBarProps) => {
-  const [title, setTitle] = useState("Calculator")
+  const [customTitle, setCustomTitle] = useState<string | null>(null)
+  const title = customTitle ?? "Calculator"
```

Comment:
Here and in the other components, instead of using React state here, store it in the `TileModel`:
```suggestion
  const title = tile.title || t("DG.DocumentController.calculatorTitle")
```
Also note that this:
1. retrieves the default title from our localized strings
2. uses `||` instead of `??` so that empty strings will be replaced with the default